### PR TITLE
`plot_time_to_collision` Add optional argument for plot save location

### PR DESCRIPTION
**Pull request by adamlm**
# PR Details
## Related Jira Key

Closes [CDAR-873](https://usdot-carma.atlassian.net/browse/CDAR-873)
[CDAR-866](https://usdot-carma.atlassian.net/browse/CDAR-866)

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[CDAR-873]: https://usdot-carma.atlassian.net/browse/CDAR-873?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CDAR-866]: https://usdot-carma.atlassian.net/browse/CDAR-866?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: vru_use_case_scripts/plot_time_to_collision [hunk 1]
```diff
@@ -125,6 +125,8 @@ parser.add_argument(
     required=True,
 )
 
+parser.add_argument("--plot-dir", type=Path, required=True)
```
Comment (paulbourelly999) [inline vru_use_case_scripts/plot_time_to_collision:128]: plot_dir vs figure_dir (see other related PRs for same comment)

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
also i think the parameter is called `plots-dir` everywhere else but here there is no s.

File: vru_use_case_scripts/plot_time_to_collision [hunk 2]
```diff
@@ -125,6 +125,8 @@ parser.add_argument(
     required=True,
 )
 
+parser.add_argument("--plots-dir", type=Path, default=Path("plots"), required=True)
```
Comment (paulbourelly999) [inline vru_use_case_scripts/plot_time_to_collision:128]: Is required=True and default not a contradiction?

**Review by paulbourelly999 — CHANGES_REQUESTED** [review]
Can you have a default value for a required argument?

File: vru_use_case_scripts/plot_time_to_collision (hunk 2 above)
Comment (adamlm) [reply vru_use_case_scripts/plot_time_to_collision]: Yes, that is a contradiction. I made it optional.

**Review by paulbourelly999 — APPROVED** [review]

**Pull request merged**